src/main/java/com/cinema/dao/impl/UserDaoImpl.java

```diff
@@ -48,7 +48,8 @@ public User add(User user) {
     public Optional<User> findByEmail(String email) {
         try (Session session = sessionFactory.openSession()) {
             Query<User> userQuery =
-                    session.createQuery("from User where email = :email", User.class);
+                    session.createQuery("from User u "
+                            + "left join fetch u.roles roles where u.email = :email", User.class);
```

Comment:
is this kind of situation is possible that you need left join fetch e.g user without roles?